### DISCUSSION TITLE
Access PR files on action

### DISCUSSION THREAD

**Discussion by ChildishGiant** [community/community#26522] [Actions]
<p>If I have an action trigger on PR, is there a way I can clone that PR to then run tests against it?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/childishgiant">@childishgiant</a> ,</p>
<p>Thank you for being here!</p>
<p>You can get the pull request files’ name via rest api, please check my answer in the <a href="https://github.community/t/get-list-of-files-on-pull-request-merge/17226">ticket</a>.</p>
<p>Could you please explain more about ‘clone PR and run tests against it’?</p>
<p>When pull_request is triggered, it will start related check runs, and if you have new commit on the source branch, the workflow will be triggered again, unless you define the types of the event, please refer to more details <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener">here.</a></p>
<p>If the current PR exists, you CANNOT create a new PR for the same source/target branches.</p>

**Comment by ChildishGiant** [comment]
<p>When a PR is submitted I want to be able to clone that PR to the instance running from an action so I can then run unit tests on it. A way to read the files in the PR. Sorry if this doesn’t make much sense?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can use the checkout action as usual (e.g. with a push event), it will automatically check out the latest commit in the pull request (unless you give another ref).</p>
